mono/metadata/threads.c

```diff
@@ -1952,42 +1973,6 @@ mono_thread_internal_current_handle (void)
 	return MONO_HANDLE_NEW (MonoInternalThread, mono_thread_internal_current ());
 }
 
-static MonoThreadInfoWaitRet
```

Comment:
This function was extracted to more easily review and implement it. Don't remove it. (PS: look at the history).

Reply:
It seems no easier to review or implement either way. The non-inline version needs a frame, to be resilient against other callers materializing, as discussed in another PR and this resolves that without growing the code. If there was another caller, or a unit test just for the function, or the function was large,  or cold vs. its one caller, those would be good reasons for keeping it separate.